src/experiment.rs

```diff
@@ -483,4 +507,39 @@ mod tests {
 
         assert!(seen);
     }
+
+    #[tokio::test]
+    async fn it_runs_experimental_result_and_ignores_control() {
+        let mut seen = false;
+        let exists = Experiment::new("test")
+            .control(async {
+                seen = true;
+                Err::<bool, &str>("failed")
+            })
+            .experimental(async { Ok::<_, &str>(true) })
+            .rollout_strategy(RolloutDecision::UseExperimental)
+            .run_result()
+            .await;
+
+        assert_eq!(exists, Ok(true));
+        // should not change seen due to only UseExperimental
+        assert!(!seen);
+    }
+
+    #[tokio::test]
+    async fn it_runs_experimental_and_ignores_control() {
+        let mut not_seen = true;
```

Comment:
nit: a little late now but it's weird that the above test calls this `seen` and this one is `not_seen`